receiver/statsdreceiver/transport/client/client.go

```diff
@@ -45,6 +45,15 @@ func (s *StatsD) connect() error {
 		if err != nil {
 			return err
 		}
+	case "unixgram":
+		unixAddr, err := net.ResolveUnixAddr(s.transport, s.address)
+		if err != nil {
+			return err
+		}
+		s.Conn, err = net.DialUnix(s.transport, nil, unixAddr)
+		if err != nil {
+			return err
+		}
```

Comment:
I would have two functions:
 - connectUDP
 - connectUnixgram

so that this one becomes more readable. Moreover I would wrap errors